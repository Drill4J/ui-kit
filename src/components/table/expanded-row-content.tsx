import * as React from 'react';
// eslint-disable-next-line import/no-cycle
import { TableRow } from './table__row';

interface Props {
  data: any;
  expandedColumns?: any[];
  idKey?: string;
  expandedRows?: string[];
  gridExpandedTemplateColumns: string;
}

export const ExpandedRowContent = ({
  data = [], expandedColumns = [], idKey = 'name', gridExpandedTemplateColumns,
}: Props) => data.map((item: any, index: number) => (
  <TableRow
    key={idKey ? String(item[idKey]) : index}
    item={item}
    columns={expandedColumns}
    index={index}
    nested
    nestedLast={index === data.length - 1}
    gridTemplateColumns={gridExpandedTemplateColumns}
  />
));
