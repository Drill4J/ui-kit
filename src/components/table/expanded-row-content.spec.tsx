import renderer from 'react-test-renderer';

import { ExpandedRowContent } from './expanded-row-content';

describe('ExpandedRowContent', () => {
  it('should match snapshot', () => {
    const tree = renderer
      .create(
        <ExpandedRowContent data={['1', '2', '3']} gridExpandedTemplateColumns="1fr 1fr 1fr" />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
