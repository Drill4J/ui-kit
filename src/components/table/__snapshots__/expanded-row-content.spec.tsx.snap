// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpandedRowContent should match snapshot 1`] = `
Array [
  <div
    className="table-row table-row--nested"
    style={
      Object {
        "backgroundColor": undefined,
        "display": "grid",
        "gridTemplateColumns": "1fr 1fr 1fr",
      }
    }
  />,
  <div
    className="table-row table-row--nested"
    style={
      Object {
        "backgroundColor": undefined,
        "display": "grid",
        "gridTemplateColumns": "1fr 1fr 1fr",
      }
    }
  />,
  <div
    className="table-row table-row--nested table-row--nested-last"
    style={
      Object {
        "backgroundColor": undefined,
        "display": "grid",
        "gridTemplateColumns": "1fr 1fr 1fr",
      }
    }
  />,
]
`;
